From fa8991b335a0671e3d15b8217239409d9de28942 Mon Sep 17 00:00:00 2001
From: Lorenzo Pirritano <lpirritano@anaconda.com>
Date: Fri, 10 Jan 2025 18:25:03 +0100
Subject: [PATCH 5/6] install headers

---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 798a1cb..8632645 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -195,3 +195,11 @@ configure_package_config_file(
 # Install the Config file
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/pybind11_protobufConfig.cmake"
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/pybind11_protobuf)
+
+# Install headers
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/pybind11_protobuf/
+  DESTINATION include/pybind11_protobuf
+  FILES_MATCHING PATTERN "*.h"
+  PATTERN "tests" EXCLUDE
+)
-- 
2.39.1

