From 3f14c5352385fe3748a234cc318b72f88fe683e1 Mon Sep 17 00:00:00 2001
From: Lorenzo Pirritano <lpirritano@anaconda.com>
Date: Fri, 20 Dec 2024 16:08:22 +0100
Subject: [PATCH 1/2] 3.13_support

---
 ortools/python/setup.py.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ortools/python/setup.py.in b/ortools/python/setup.py.in
index ff63622..fd3a16e 100644
--- a/ortools/python/setup.py.in
+++ b/ortools/python/setup.py.in
@@ -147,6 +147,7 @@ setup(
         'Programming Language :: Python :: 3.10',
         'Programming Language :: Python :: 3.11',
         'Programming Language :: Python :: 3.12',
+        'Programming Language :: Python :: 3.13',
         'Programming Language :: C++',
         'Programming Language :: Python :: Implementation :: CPython',
         'Topic :: Office/Business :: Scheduling',
-- 
2.39.1

