diff -ur ortools/base/logging.h ortools_new/base/logging.h
--- ortools/base/logging.h	2023-03-10 08:19:51.000000000 +0000
+++ ortools_new/base/logging.h	2024-11-06 15:48:36.297386978 +0000
@@ -51,16 +51,4 @@
   GLOG_FATAL = static_cast<int>(absl::LogSeverity::kFatal),
 };
 }  // namespace google
-
-// Implementation of the `AbslStringify` interface. This adds `DebugString()`
-// to the sink. Do not rely on exact format.
-namespace google {
-namespace protobuf {
-template <typename Sink>
-void AbslStringify(Sink& sink, const Message& msg) {
-  sink.Append(msg.DebugString());
-}
-}  // namespace protobuf
-}  // namespace google
-
 #endif  // OR_TOOLS_BASE_LOGGING_H_
Only in ortools/base: logging.h_new
